addons/mail/models/discuss/res_partner.py

```diff
@@ -21,29 +21,31 @@ def _get_channels_as_member(self):
         self.ensure_one()
         channels = self.env["discuss.channel"]
         # get the channels and groups
-        channels |= self.env["discuss.channel"].search(
-            [
-                ("channel_type", "in", ("channel", "group")),
-                ("channel_partner_ids", "in", [self.id]),
-            ]
-        )
-        # get the pinned direct messages
-        channels |= self.env["discuss.channel"].search(
-            [
-                ("channel_type", "=", "chat"),
-                (
-                    "channel_member_ids",
-                    "in",
-                    self.env["discuss.channel.member"]
-                    .sudo()
-                    ._search(
-                        [
-                            ("partner_id", "=", self.id),
-                            ("is_pinned", "=", True),
-                        ]
-                    ),
+        domain_channels_groups = [
+            ("channel_type", "in", ("channel", "group")),
+            ("channel_partner_ids", "in", [self.id]),
+        ]
+        # get the channels which are pinned and not of type channel or group
+        domain_non_channel_or_groups = [
+            ("channel_type", "not in", ("channel", "group")),
+            (
+                "channel_member_ids",
+                "in",
+                self.env["discuss.channel.member"]
+                .sudo()
+                ._search(
+                    [
+                        ("partner_id", "=", self.id),
+                        ("is_pinned", "=", True),
+                    ]
                 ),
-            ]
+            ),
+        ]
+        channels = self.env["discuss.channel"].search(
+            expression.OR([
```

Comment:
To be double checked but I think making OR here makes the query much slower than having 2 separate queries and joining results.

Reply:
I believe the only fast way to make a single query would be to manually make a double outer join (similarly to what is done to fetch all members in one go in `discuss_channel_member`).

Reply:
For caution, I would probably not change anything in this PR, and keep 2 separate queries. (only removing overrides is fine)

Reply:
okay